internal/controllers/machineset/machineset_preflight.go

```diff
@@ -173,7 +175,12 @@ func (r *Reconciler) kubernetesVersionPreflightCheck(cpSemver, msSemver semver.V
 	if msSemver.Minor > cpSemver.Minor {
 		return pointer.String(fmt.Sprintf("MachineSet version (%s) and ControlPlane version (%s) do not conform to the kubernetes version skew policy as MachineSet version is higher than ControlPlane version (%q preflight failed)", msSemver.String(), cpSemver.String(), clusterv1.MachineSetPreflightCheckKubernetesVersionSkew))
 	}
-	if msSemver.Minor < cpSemver.Minor-2 {
+	minorSkew := uint64(3)
```

Comment:
the godoc in l.173 is slightly off, let's replace that with something generic like "cannot be outside of the supported skew"

Reply:
Adjusted to inject the `minorSkew` variable instead of the hardcoded 2.